### ISSUE TITLE
Multiselect dropdown item missing aria-label attribute

### ISSUE THREAD

**Issue by fildahui** [microsoft/fluentui#18313]
The current version of the Dropdown (when multi selection is enabled) does not render the `aria-label` attribute on the checkbox input anymore.
The attribute was there with version 7 of the library.

This is the rendered element (from fluentui sample)
```
<input
   class="input-336"
   type="checkbox"
   aria-selected="true"
   role="option"
   data-index="1"
   data-is-focusable="true"
   id="Dropdown319-list1"
   title="Apple"
   aria-posinset="1"
   aria-setsize="7"
   aria-checked="true"
   data-ktp-execute-target="true"
   checked=""
   tabindex="0"
>
```

### Environment Information

- **Package version(s)**: 8.14.14

### Please provide a reproduction of the bug in a codepen:

The issue is present in the Dropdown samples page (multiselection sample section):

[https://developer.microsoft.com/en-us/fluentui#/controls/web/dropdown](https://developer.microsoft.com/en-us/fluentui#/controls/web/dropdown)

Just inspect the element once it is open

#### Actual behavior:

`aria-label` attribute is missing from the checkbox input

#### Expected behavior:

`aria-label` should be present on the input as it was before v. 8

### Priorities and help requested:

Are you willing to submit a PR to fix? No

Requested priority: Normal

Products/sites affected: (if applicable)

**Comment by TristanWatanabe** [comment]
This was intentionally done when `Checkbox` was updated by #18260 as having the the item's label be the fallback for the `aria-label` of the checkbox was bad practice. Multi-select `Dropdown` will now need to specifically pass an `ariaLabel` prop to `Checkbox` now though and I'll work on getting a PR out for that.
